File: src/com/limelight/nvstream/NvConnectionListener.java
package com.limelight.nvstream;

public interface NvConnectionListener {
	
	public enum Stage {
		LAUNCH_APP("app"),
		HANDSHAKE("handshake"),
		CONTROL_START("control connection"),
		VIDEO_START("video stream"),
		AUDIO_START("audio stream"),
		CONTROL_START2("control connection"),
		INPUT_START("input connection");
		
		private String name;
		private Stage(String name) {
			this.name = name;
		}
		
		public String getName() {
			return name;
		}
	};
	
	public void stageStarting(Stage stage);
	public void stageComplete(Stage stage);
	public void stageFailed(Stage stage);
	
	public void connectionStarted();
	public void connectionTerminated(Exception e);
}

File: app/src/main/java/com/limelight/nvstream/NvConnectionListener.kt
package com.limelight.nvstream

interface NvConnectionListener {
    fun stageStarting(stage: String)
    fun stageComplete(stage: String)
    fun stageFailed(stage: String, portFlags: Int, errorCode: Int)

    fun connectionStarted()
    fun connectionTerminated(errorCode: Int)
    fun connectionStatusUpdate(connectionStatus: Int)

    fun displayMessage(message: String)
    fun displayTransientMessage(message: String)

    fun rumble(controllerNumber: Short, lowFreqMotor: Short, highFreqMotor: Short)
    fun rumbleTriggers(controllerNumber: Short, leftTrigger: Short, rightTrigger: Short)
    fun setAdaptiveTriggers(
        controllerNumber: Short,
        eventFlags: Byte,
        typeLeft: Byte,
        typeRight: Byte,
        left: ByteArray,
        right: ByteArray
    )

    fun setHdrMode(enabled: Boolean, hdrMetadata: ByteArray?)

    fun setMotionEventState(controllerNumber: Short, motionType: Byte, reportRateHz: Short)

    fun setControllerLED(controllerNumber: Short, r: Byte, g: Byte, b: Byte)

    fun onResolutionChanged(width: Int, height: Int)

    fun onCursorUpdate(
        flags: Int,
        shapeId: Int,
        width: Int,
        height: Int,
        hotspotX: Int,
        hotspotY: Int,
        bgraPixels: ByteArray?
    )
}
